### ISSUE TITLE
Cannot update via app, must download and install again

### ISSUE THREAD

**Issue by wallym** [Microsoft/sqlopsstudio#732]
- SQL Operations Studio Version: 25.7 (iirc)
Not earth-shattering, but did want to at least report this issue.  It might rear its head again on the next update.

Steps to Reproduce:
with the January build, trying to download and update within the app results in:
1.  a cutoff error message.
2. the ability to update within the application does not work.

![screen shot 2018-02-19 at 9 01 33 am](https://user-images.githubusercontent.com/366611/36381699-725dc0c6-1554-11e8-8ec8-3105fce79823.png)

**Comment by jtsom** [comment]
Same problem here on OS X High Sierra. Error message states:

Cannot update while running on a read-only volume. The application is on a read-only volume. Please move the application and try again. If you're on macOS Sierra or later, you'll need to move the application out of the Downloads directory. See https://github.com/Squirrel/Squirrel.Mac/issues/182 for more information.

However, the application is in the /Applications folder, not in the Downloads folder.

**Comment by joeangry** [comment]
This seems disabled for OSX purposely.
https://github.com/Microsoft/sqlopsstudio/blob/master/src/vs/code/electron-main/menus.ts#L1032

**Comment by jtsom** [comment]
No, I think that's just not adding the "Check for update" to the Mac's Help menu. On the Mac its under the application's menu.

**Comment by kburtram** [comment]
I saw this error when trying to auto-update when running SQL Ops from the Downloads folder, as mentioned above.  When I move SQL Ops to Applications and restarted, I was able to successfully auto-update.

As a workaround the latest build is available for direct download under GitHub Releases (currently https://github.com/Microsoft/sqlopsstudio/releases/download/0.26.6/sqlops-macos-0.26.7.zip).

If SQL Ops is already in the Applications directory I'm not sure why GateKeeper would still be sandboxing the app.  I'll try to search around for more details about this.  Any additional repro steps would help?  Do you know if VS Code auto-update is working on this system?

I also have a macOS .pkg installer package available.  Perhaps if SQL Ops is installed using that auto-update will work as expected in the repro environment.  I'll post that installer file in a bit.

**Comment by kburtram** [comment]
It looks like when a macOS is downloaded from the internet it will automatically be sandboxed until the application is moved due to an OS feature call `app translocation`.  That's why if the app is download to Downloads and then moved to Applications it is updatable since it's not sandboxed.  When the app is autoupdated it gets sandboxed again until moved again.

A workaround is to move the app, for example to desktop, using Command+Drag.  Then move the app back to original location.  This will un-sandbox the app and allow auto-update to succeed.  Unfortunately, this will likely be a common issue with March release since most users were probably sandboxed by the Feb auto-update.  

I'll look into how VS Code resolves this situation and try to get an updated hotfix build.  Though I'm not sure what we'll be able to do regarding machines in the sandboxed state other than the manual workaround.

**Comment by MattIrv** [comment]
We'll need to fix this for new releases in the future by running the `xattr -d com.apple.quarantine /Applications/SQL\ Operations\ Studio.app/` command before uploading the release. I just uploaded a new zip for 0.26.7 that has the quarantine attribute removed and I've verified that this fixes it for anyone using the auto updater going forward.

Unfortunately you'll still run into this if you updated already, so if you run into the "Cannot update while running on a read-only volume" error you can use one of the workarounds in the comment above.

**Comment by pohvak** [comment]
app is in Applications folder when "check for update" is pressed it says there are no new versions, but I know theres new version (I got 0.29.3 now and 0.30 has been already released

**Comment by MattIrv** [comment]
@pohvak The latest public preview release is 0.29.3, so it sounds like everything is behaving correctly. We've released a few insiders builds since then, but public preview releases don't auto update to insiders builds since they're less stable.

If you want insiders builds you can download [the latest insiders build](https://github.com/Microsoft/sqlopsstudio/releases/tag/0.30.3) and then you'll get automatic updates for new insiders builds in the future too.
